workflow/marshal.go

```diff
@@ -0,0 +1,145 @@
+package workflow
```

Comment:
I don't see where the MarshalJSON is used, can you point me

Reply:
The marshal is just on the nodes, not on the graph, but the unmarshal is on the graph and create the specific nodes.

The marshal is used in the database because we still encode everything in json in the db

Reply:
> The marshal is used in the database because we still encode everything in json in the db

👍 . So why do we need custom marhsal/unmurshal in that case?

Reply:
json is not typed so when we marshal the data we have a json with the data of each node but when we unmarshal we don't know what type of node it is.
The marshal/unmarshal here adds the type in the serialization.
https://github.com/mesg-foundation/engine/pull/1258/files#diff-f46b0dfcf69190d0795e825675bfdf5aR13 and https://github.com/mesg-foundation/engine/pull/1258/files#diff-f46b0dfcf69190d0795e825675bfdf5aR97